test/processStates_test.js

```diff
@@ -0,0 +1,261 @@
+goog.module('datalayerhelper.helper.testing.processStates');
+goog.setTestOnly();
+
+const {DataLayerHelper, buildAbstractModelInterface_} = goog.require('helper');
+
+describe('The processStates function', function() {
+  /**
+   * Asserts that calling processStates_ with the given arguments will result
+   * in the expected model state and expected calls to the listener.
+   *
+   * @param {Array<Object>} states The states argument for the call to
```

Comment:
Please make sure that nullability and template parameter types are explicitly specified
https://google.github.io/styleguide/jsguide.html#jsdoc-nullability
https://google.github.io/styleguide/jsguide.html#jsdoc-template-parameter-types